podvm/Dockerfile.podvm

```diff
@@ -0,0 +1,22 @@
+# Copyright Confidential Containers Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Builds pod vm image inside container
+#
+ARG BUILDER_IMG="quay.io/confidential-containers/cloud-api-adaptor-podvm-builder"
+
+FROM ${BUILDER_IMG} AS podvm_builder
+
+ARG CLOUD_PROVIDER=libvirt
+
+# Building binaries separately allows for reuse in case of error
+# with image building step
+RUN cd cloud-api-adaptor/podvm && \
+     CLOUD_PROVIDER=$CLOUD_PROVIDER make binaries
+
+RUN cd cloud-api-adaptor/podvm && \
+     CLOUD_PROVIDER=$CLOUD_PROVIDER make image
```

Comment:
@whitebrandy let's handle the rhel (including centos image) generation using container builds in a separate PR for the following reasons:
1. We need to figure out a mechanism which avoids copying any files from local host directory and I think it will need some discussions.
2. Use a RHEL or CentOS based builder image for building the dynamic linked binaries or installing the native packages